### PR TITLE
[unauthorizedtv] Add new extractor

### PR DESCRIPTION
**Pull request by kevgrig**
### Before submitting a *pull request* make sure you have:
- [X] At least skimmed through [adding new extractor tutorial](https://github.com/ytdl-org/youtube-dl#adding-support-for-a-new-site) and [youtube-dl coding conventions](https://github.com/ytdl-org/youtube-dl#youtube-dl-coding-conventions) sections
- [X] [Searched](https://github.com/ytdl-org/youtube-dl/search?q=is%3Apr&type=Issues) the bugtracker for similar pull requests
- [X] Checked the code with [flake8](https://pypi.python.org/pypi/flake8)

### In order to be accepted and merged into youtube-dl each piece of code must be in public domain or released under [Unlicense](http://unlicense.org/). Check one of the following options:
- [X] I am the original author of this code and I am willing to release it under [Unlicense](http://unlicense.org/)
- [ ] I am not the original author of this code but it is in public domain or released under [Unlicense](http://unlicense.org/) (provide reliable evidence)

### What is the purpose of your *pull request*?
- [ ] Bug fix
- [ ] Improvement
- [X] New extractor
- [ ] New feature

---

### Description of your *pull request* and other information

Extractor for https://www.unauthorized.tv/. Example video at https://www.unauthorized.tv/programs/owens-shorts?cid=231148.

### CODE REVIEW

File: youtube_dl/extractor/unauthorizedtv.py [hunk 1]
```diff
@@ -0,0 +1,80 @@
...
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        cid = None
+
+        if "?cid=" in video_id:
+            cid = int(video_id[video_id.find('=') + 1:])
+
+        if self._downloader.params.get('verbose', False):
+            print(video_id)
```
Comment (dstftw) [inline youtube_dl/extractor/unauthorizedtv.py]: Remove.

File: youtube_dl/extractor/unauthorizedtv.py [hunk 2]
```diff
@@ -0,0 +1,80 @@
...
+        }
+
+        chaptersJson = self._download_json(
+            'https://www.unauthorized.tv/api/contents/%s' % video_id,
+            video_id,
+            headers=headers
+        )
+
+        if self._downloader.params.get('verbose', False):
+            print(chaptersJson)
```
Comment (dstftw) [inline youtube_dl/extractor/unauthorizedtv.py]: Remove.

File: youtube_dl/extractor/unauthorizedtv.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
...
+        else:
+            for item in metadata:
+                if item["id"] == cid:
+                    video_title = item['title']
+                    video_url = item['subject']['versions']['hls']
+
+        if self._downloader.params.get('verbose', False):
+            print(metadata)
+            print(video_title)
+            print(video_url)
```
Comment (dstftw) [inline youtube_dl/extractor/unauthorizedtv.py]: Remove.

File: youtube_dl/extractor/unauthorizedtv.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        chaptersJson = self._download_json(
+            'https://www.unauthorized.tv/api/contents/%s' % video_id,
+            video_id,
+            headers=headers
+        )
+
+        if self._downloader.params.get('verbose', False):
+            print(chaptersJson)
+
+        chapters = '&ids[]='.join([str(x) for x in chaptersJson['chapters']])
```
Comment (dstftw) [inline youtube_dl/extractor/unauthorizedtv.py]: compat_str

File: youtube_dl/extractor/unauthorizedtv.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+            'title': 'Millennials',
+        }
+    }
+
+    def _real_extract(self, url):
+        video_id = self._match_id(url)
+        cid = None
+
+        if "?cid=" in video_id:
+            cid = int(video_id[video_id.find('=') + 1:])
```
Comment (dstftw) [inline youtube_dl/extractor/unauthorizedtv.py]: Capture with regex in `_VALID_URL`.

File: youtube_dl/extractor/unauthorizedtv.py [hunk 6]
```diff
@@ -0,0 +1,80 @@
...
+        if self._downloader.params.get('verbose', False):
+            print(video_id)
+
+        html = self._download_webpage(url, video_id)
+
+        csrf_token = self._html_search_meta(
+            'csrf-token',
+            html,
+            'csrf token',
+            default=None
```
Comment (dstftw) [inline youtube_dl/extractor/unauthorizedtv.py]: Move on a single line.

File: youtube_dl/extractor/unauthorizedtv.py [hunk 7]
```diff
@@ -0,0 +1,80 @@
...
+        )
+
+        headers = {
+            'Referer': url,
+            'X-Requested-With': 'XMLHttpRequest',
+            'X-CSRF-Token': csrf_token,
```
Comment (dstftw) [inline youtube_dl/extractor/unauthorizedtv.py]: May be None.

File: youtube_dl/extractor/unauthorizedtv.py [hunk 8]
```diff
@@ -0,0 +1,80 @@
...
+                    video_url = item['subject']['versions']['hls']
+
+        if self._downloader.params.get('verbose', False):
+            print(metadata)
+            print(video_title)
+            print(video_url)
+
+        return {
+            'id': video_id,
+            'title': video_title,
```
Comment (dstftw) [inline youtube_dl/extractor/unauthorizedtv.py]: Possibly referenced before assignment.

File: youtube_dl/extractor/unauthorizedtv.py [hunk 9]
```diff
@@ -0,0 +1,80 @@
...
+            'url': video_url,
```
Comment (dstftw) [inline youtube_dl/extractor/unauthorizedtv.py]: Possibly referenced before assignment.

**Comment by kevgrig** [comment]
@dstftw Changes made. Please re-review when you have a chance.

**Comment by kevgrig** [comment]
Closing this because the service now requires a login. For anybody that wants to manually apply this patch: https://github.com/kevgrig/youtube-dl/blob/unauthorizedtv/youtube_dl/extractor/unauthorizedtv.py

**Pull request closed without merging**